include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -0,0 +1,585 @@
+ * needed to provide the range-based functionality of the file.
+ *
+ * **None of the fields are required** when writing but will be defaulted
+ * to '0' for numeric fields and '*' for other fields.
+ *
+ * ### SAM format columns -> fields
+ *
+ * As many users will be accustomed to the columns of the SAM format, here is a
+ * mapping of the common SAM format columns to the SeqAn3 record fields:
+ *
+ * | #  | SAM Column ID |  FIELD name                                       |
+ * |:--:|:--------------|:--------------------------------------------------|
+ * | 1  | QNAME         | **seqan3::field::ID**                             |
+ * | 2  | FLAG          | seqan3::field::FLAG                               |
+ * | 3  | RNAME         | **seqan3::field::REF_ID**                         |
+ * | 4  | POS           | **seqan3::field::REF_OFFSET**                     |
+ * | 5  | MAPQ          | seqan3::field::MAPQ                               |
+ * | 6  | CIGAR         | implicilty stored in **seqan3::field::ALIGNMENT** |
+ * | 7  | RNEXT         | seqan3::field::MATE (tuple pos 0)                 |
+ * | 8  | PNEXT         | seqan3::field::MATE (tuple pos 1)                 |
+ * | 9  | TLEN          | seqan3::field::MATE (tuple pos 2)                 |
+ * | 10 | SEQ           | seqan3::field::SEQ                                |
+ * | 11 | QUAL          | seqan3::field::QUAL                               |
+ *
+ * The (read sequence/query) **OFFSET** will be required to store the soft
+ * clipping information at the read start (end clipping will be automatically
+ * deduced by how much the read sequence length + offset is larger than the
+ * alignment length).
+ *
+ * \attention Hard clipping information in the SAM file will be lost and
+ *            cannot be recovered. As Hard clipping is usually a quality cut off
+ *            and the respective sequence part is not present in the SAM file,
+ *            we expect that this information is in 99% of the cases irrelevant.
+ *            If requested, the hard clipping information could be stored in a
+ *            separate field later on.
+ *
+ * ### Format Check
+ *
+ * The format checks are implemented according to the official
+ * [SAM format specializations](https://samtools.github.io/hts-specs/SAMv1.pdf)
+ * in order to ensure correct SAM file output.
+ *
+ * If any error is detected, a seqan3::format_error is thrown. There is only
+ * **one exception**: If the sequences like `id`, `ref_id`, `seq`, etc. ...
+ * contain any illegal white space character (seqan3::is_space) they are truncated
+ * up until its occurrence instead of throwing an error. This avoids severe
+ * format errors for the tab separated SAM format.
+ *
+ * ### Header implementation
+ *
+ * The header_ptr (if set in the seqan3::alignment_file_output or given by a pointer
+ * in the seqan3::field::HEADER field) is only printed once in the beginning
```

Comment:
the header_pointer is given by a pointer? Sounds kind of weird. Since all of the discussed logic is part of the file, I would explain it there and just mention here that the SAM header is automatically written with the first record based on the information stored in the alignment_file.